File: qwencoder-eval/instruct/eval-dev-quality/testdata/golang/light/matchBytes.go
package light

func matchBytes(s1 []byte, s2 []byte) bool {
	for i := 0; i < len(s1); i++ {
		c1 := s1[i]
		c2 := s2[i]

		if c1 != c2 {
			c1 |= 'a' - 'A'
			c2 |= 'a' - 'A'

			if c1 != c2 || c1 < 'a' || c1 > 'z' {
				return false
			}
		}
	}

	return true
}

File: qwencoder-eval/instruct/eval-dev-quality/testdata/java/light/src/main/java/com/eval/MatchBytes.java
package com.eval;

class MatchBytes {
    static boolean matchBytes(byte[] s1, byte[] s2) {
		for (int i = 0; i < s1.length; i++) {
			byte c1 = s1[i];
			byte c2 = s2[i];

			if (c1 != c2) {
				c1 |= 'a' - 'A';
				c2 |= 'a' - 'A';

				if (c1 != c2 || c1 < 'a' || c1 > 'z') {
					return false;
				}
			}
		}

		return true;
	}
}
